debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala

```diff
@@ -472,6 +492,28 @@ object Parse {
           }
       }
 
+    // Parse the root name field from the launch config
+    // Defaults to None
+    //
+    // arguments: Launch config
+    def parseRootName(arguments: JsonObject) =
+      Option(arguments.getAsJsonPrimitive("rootName"))
+        .map(_.getAsString())
+        .getOrElse("")
+        .asRight[String]
+        .toEitherNel
```

Comment:
This doesn't actually default to `None`, it defaults to `""`, which I think we should avoid. Instead we can return an `Either[String, Option[String]]` and pass that along, so downstream code doesn't have to check for the empty string. Something like:
```suggestion
    // Defaults to None
    //
    // arguments: Launch config
    def parseRootName(arguments: JsonObject): Either[String, Option[String]] =
      Right(
        Option(arguments.getAsJsonPrimitive("rootName"))
          .map(_.getAsString())
      ).toEitherNel
```

Reply:
@arosien So there a way to make it just `Option[String]` instead of `Either[String, Option[String]]`? I had tried to do that before but it caused errors with the `parMapN` functions

Reply:
When I add this and change all `rootName` to `Either[String, Option[String]]` I get a bunch of errors like below:

```bash
[error]  found   : (java.nio.file.Path, java.nio.file.Path, Boolean, String, org.apache.daffodil.debugger.dap.Parse.Debugee.LaunchArgs.InfosetOutput, Either[String,Option[String]], Either[String,Option[String]], Map[String,String], Map[String,String]) => org.apache.daffodil.debugger.dap.Parse.Debugee.LaunchArgs.Manual
[error]  required: (java.nio.file.Path, java.nio.file.Path, Boolean, String, Product with Serializable with org.apache.daffodil.debugger.dap.Parse.Debugee.LaunchArgs.InfosetOutput, Option[String], Option[String], scala.collection.immutable.Map[String,String], scala.collection.immutable.Map[String,String]) => ?
[error]         ).parMapN(LaunchArgs.Manual.apply)
```

Which are similar errors I got when trying to just do `Option[String]` and was the reason I went with `String` then just to do that matching before passing to `compileFile`

Reply:
The signature of `LaunchArgs.Manual.apply` needs to be updated from taking `String` to taking `Option[String]`.

Reply:
(same changes for `parseNamespace` since it is also optional)

Reply:
If you can't get it I'll push something up.

Reply:
@arosien So I tried that but I am getting these errors:

```bash
[error] /Users/sdell/apache/daffodil-vscode/debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala:295:18: could not find implicit value for parameter p: cats.NonEmptyParallel[[+A]scala.util.Either[java.io.Serializable,A]]
[error]         ).parMapN(LaunchArgs.Manual.apply)
[error]                  ^
[error] /Users/sdell/apache/daffodil-vscode/debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala:326:20: could not find implicit value for parameter p: cats.NonEmptyParallel[[+A]scala.util.Either[java.io.Serializable,A]]
[error]           ).parMapN(LaunchArgs.Manual.apply)
[error]                    ^
[error] /Users/sdell/apache/daffodil-vscode/debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala:343:24: could not find implicit value for parameter p: cats.NonEmptyParallel[[+A]scala.util.Either[java.io.Serializable,A]]
[error]               ).parMapN(LaunchArgs.TDMLConfig.Generate.apply)
[error]                        ^
[error] /Users/sdell/apache/daffodil-vscode/debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala:358:24: could not find implicit value for parameter p: cats.NonEmptyParallel[[+A]scala.util.Either[java.io.Serializable,A]]
[error]               ).parMapN(LaunchArgs.TDMLConfig.Append.apply)
[error]                        ^
[error] /Users/sdell/apache/daffodil-vscode/debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala:371:24: could not find implicit value for parameter p: cats.NonEmptyParallel[[+A]scala.util.Either[java.io.Serializable,A]]
[error]               ).parMapN(LaunchArgs.TDMLConfig.Execute.apply)
[error]                        ^
[error] /Users/sdell/apache/daffodil-vscode/debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala:503:9: polymorphic expression cannot be instantiated to expected type;
[error]  found   : [A, AA >: A]cats.data.EitherNel[AA,Option[String]]
[error]     (which expands to)  [A, AA >: A]scala.util.Either[cats.data.NonEmptyList[AA],Option[String]]
[error]  required: Either[String,Option[String]]
[error]       ).toEitherNel
[error]         ^
[error] /Users/sdell/apache/daffodil-vscode/debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala:513:9: polymorphic expression cannot be instantiated to expected type;
[error]  found   : [A, AA >: A]cats.data.EitherNel[AA,Option[String]]
[error]     (which expands to)  [A, AA >: A]scala.util.Either[cats.data.NonEmptyList[AA],Option[String]]
[error]  required: Either[String,Option[String]]
[error]       ).toEitherNel
```

Reply:
Try merging in this PR: https://github.com/shanedell/daffodil-vscode/pull/120

Reply:
So @arosien with your changes what do I send over as `rootNamespace` from the extension to the debugger so that it is `None`. Currently if I send `""` it will be `""` which causes an error as this defaults the namespace to `{No_Namespace}`, trying to send `null` causes an error

Reply:
So this actually only happens if you specify in the launch.json `"rootNamespace": ""` otherwise it works as expected. Maybe this how it should function also in general right based of on how Daffodil is handling it in the backend?